resources/ext/config/conf.d/puppetserver.conf

```diff
@@ -4,10 +4,14 @@ jruby-puppet: {
     # Puppet server expects to load Puppet from this location
     ruby-load-path: [/opt/puppetlabs/puppet/lib/ruby/vendor_ruby]
 
-    # This setting determines where JRuby will look for gems.  It is also
-    # used by the `puppetserver gem` command line tool.
+    # This setting determines where JRuby will install gems.  It is used for loading gems,
+    # and also by the `puppetserver gem` command line tool.
     gem-home: /opt/puppetlabs/server/data/puppetserver/jruby-gems
 
+    # This setting defines the complete "GEM_PATH" for jruby.  If set, it should include
+    # the gem-home directory as well as any other directories that gems can be loaded
+    # from (including the vendored gems directory for gems that ship with puppetserver)
+    gem-path: ${jruby-puppet.gem-home}":/opt/puppetlabs/server/data/puppetserver/vendored-jruby-gems"
```

Comment:
So one thing I just realized here is that I set this up as a single string value, because that's what the underlying JRuby setup is going to want.  But that seems kind of dumb and inconsistent since we use an array for ruby-load-path above.  Seems like it would be much better UX for me to change this to an array in the config file, and then munge it into a string programatically before I send it to JRuby.  Anyone disagree?